src/Data/Detection/DetectionRepository.cs

```diff
@@ -26,5 +26,20 @@ public IEnumerable<MDetection> ByParameter() => _context.Set<EDetection>()
                 Timestamp = DateTime.UtcNow,
             },
         };
+
+        public MDetection Insert(MCreateDetection input)
+        {
+            var result = _context.Detections.Add(DetectionFactory.MakeEntity(input));
+            _context.SaveChanges();
+            return DetectionFactory.MakeModel(result.Entity);
+        }
+
+        public IEnumerable<MDetection> ByInput(MCreateDetection detection)
+        {
+            return _context.Set<EDetection>()
+                .Where(entity => detection.Class == entity.Class && detection.Score == entity.Score)
+                .ToList()
+                .Select(DetectionFactory.MakeModel);
+        }
```

Comment:
Should we convert this method into a one-liner?

Reply:
done